src/l500/l500-device.cpp

```diff
@@ -696,11 +696,14 @@ namespace librealsense
     {
         std::string fw_version = extract_firmware_version_string((const void*)image.data(), image.size());
 
-        auto it = ivcam2::device_to_fw_min_version.find(_pid);
-        if (it == ivcam2::device_to_fw_min_version.end())
-            throw std::runtime_error("Minimum firmware version has not been defined for this device!");
+        auto min_max_fw_it = ivcam2::device_to_fw_min_max_version.find(_pid);
+        if (min_max_fw_it == ivcam2::device_to_fw_min_max_version.end())
+            throw std::runtime_error("Min and Max firmware versions have not been defined for this device!");
+
+        // Limit L515 to FW versions within the 1.5.1.3-1.99.99.99 range to differenciate from the other products
+        return (firmware_version(fw_version) >= firmware_version(min_max_fw_it->second.first)) &&
+               (firmware_version(fw_version) <= firmware_version(min_max_fw_it->second.second));
```

Comment:
I think we can use [) for ranges here so it's be `<` rather than `<=`?
Then you wouldn't have to write numbers like `1.99.99.99`...

I also don't like putting the build number in there. It's got nothing to do with the version number.

Reply:
The max is required to prevent recognizing D400's 5.12.14.50 as a valid candidate for L515.
Since no formal definition is given 1.99..... is the version that will be currently supported.

Reply:
I'm saying rather than `<= 1.99.99.99` we should say `< 2.0`